src/Classes/ServiceAPI/MyRadio_Season.php

```diff
@@ -317,16 +317,18 @@ public static function create($params = [])
         return $newSeason;
     }
 
-    public static function getForm()
+    public static function getForm($num_weeks, $startdate)
     {
         //Set up the weeks checkboxes
         $weeks = [];
-        for ($i = 1; $i <= 10; ++$i) {
+        $date = $startdate;
+        for ($i = 1; $i <= $num_weeks; ++$i) {
             $weeks[] = new MyRadioFormField(
                 'wk'.$i,
                 MyRadioFormField::TYPE_CHECK,
-                ['label' => 'Week '.$i, 'required' => false]
+                ['label' => 'Week beginning '.date("Y-m-d",$date), 'required' => false]
```

Comment:
maybe see how it goes, i wouldn't be suprised if people would end up prefering 
- freshers week
-  teaching week 1
- teaching week 2
etc etc

Reply:
That is possible but it'd be a lot more work and I doubt anyone will strongly care.

Reply:
I disagree - people refer to weeks by their number enough that I think it's worth having it there. I'd be happy to get this merged as an initial step, but that risks us forgetting it down the line.

Reply:
I also think a lot of people have no idea what the weeks mean. If you are an alumni, on a leave of absence of *cough* year in industry then booking shows with the week numbers is a huge pain in the backside.

Reply:
i shall now propose: both

Reply:
Which is what I suggested upthread:

> Or perhaps something like
> 
> freshers week (w/c 18/09)
> week 1 (wc 25/09)
> week 2 (02/10)
> etc